src/Servers/IIS/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -252,7 +252,21 @@ IN_PROCESS_APPLICATION::ExecuteApplication()
 
         if (m_pConfig->QueryCallStartupHook())
         {
-            RETURN_IF_NOT_ZERO(context->m_hostFxr.SetRuntimePropertyValue(DOTNETCORE_STARTUP_HOOK, ASPNETCORE_STARTUP_ASSEMBLY));
+            PWSTR startupHookValue;
+            // Will get property not found if the enviroment variable isn't set.
+            context->m_hostFxr.GetRuntimePropertyValue(DOTNETCORE_STARTUP_HOOK, &startupHookValue);
```

Comment:
Can this call fail?

Reply:
IMO it's better to ignore the return value than check it. If it does fail, I'd rather not block starting the application.

As the comment mentioned, if the property isn't set, GetRuntimePropertyValue will return an error for PROPERTY_NOT_FOUND. It can fail in a couple of other ways from what I can tell (https://github.com/dotnet/core-setup/blob/5cd0245860335e0b307f6a46b99c6f9244887318/src/corehost/cli/fxr/hostfxr.cpp#L663), but these errors would most likely show up by previous calls to the hosting apis. Also, I'd rather not hard code an int check as that could be risky if hostfxr starts returning another integer value for this kind of failure (I doubt it would happen, but I don't think it's worth the risk).

Reply:
The problem is that currently if an error is returned the output value is undefined see: https://github.com/dotnet/core-setup/issues/8860
It "might" work if you preinitialize it to NULL, but even with that it's kind of a fragile solution. I think we should simply check for success - otherwise set the startuphook to an empty string.
Once the above mentioned issue is fixed, the code might work as is.. but we're not there yet.